### PR TITLE
[SYM-4248] Update GitHub dynamic targets example to use prefetch

### PR DESCRIPTION
**Pull request by llam15**
# Summary
- Updates the GitHub dynamic targets example to make the `repo_name` Prompt Field `prefetch = True`
- Implements a prefetch function that shows how to make an API request and return a list of FieldOptions

https://user-images.githubusercontent.com/10479740/200004614-e1a73240-260c-4fb4-a24c-5a3c267242c3.mov

### CODE REVIEW

**Comment by llam15** [comment]
Current dependencies on/for this PR:
* main
  * **PR #49** <a href="https://app.graphite.dev/github/pr/symopsio/examples/49" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/symopsio/examples/49?utm_source=stack-comment).

File: github_dynamic_targets/impl.py [hunk 1]
```diff
@@ -9,3 +11,25 @@ def get_approvers(event):
...
+
+
+# Prefetch reducers return a list of FieldOptions that will be used to populate
+# the Slack select menu for the Prompt Field with the given "field_name"
+@prefetch(field_name="repo_name")
+def get_repos(event):
+    github_token = event.flow.environment.integrations["github"].settings["api_token_secret"].retrieve_value()
+    headers = {"Authorization": f"Token {github_token}"}
+
+    # Use the GitHub API to get the first 100 public repos in the Symopsio organization
```
Comment (Arianna2028) [inline github_dynamic_targets/impl.py:23]:
```suggestion
    # Use the GitHub API to get the first 100 public repos in the sym-test organization
```

Comment (Arianna2028) [inline github_dynamic_targets/impl.py:23]: We might want to update this to handle pagination? So it's more realistic.

**Review by Arianna2028 — COMMENTED** [review]
This is awesome! Couple notes.

File: github_dynamic_targets/impl.py [hunk 2]
```diff
@@ -9,3 +11,40 @@ def get_approvers(event):
...
+def get_repos(event):
+    github_token = event.flow.environment.integrations["github"].settings["api_token_secret"].retrieve_value()
+    headers = {"Authorization": f"Token {github_token}"}
+
+    all_options = []
+    per_page = 10
+    page = 1
+
+    while True:
+        # Use the GitHub API to get the first 100 private repos in the Sym Test organization
```
Comment (Arianna2028) [inline github_dynamic_targets/impl.py:28]: not 100 anymore

File: github_dynamic_targets/impl.py [hunk 3]
```diff
@@ -9,3 +11,40 @@ def get_approvers(event):
...
+
+# Prefetch reducers return a list of FieldOptions that will be used to populate
+# the Slack select menu for the Prompt Field with the given "field_name"
+@prefetch(field_name="repo_name")
+def get_repos(event):
+    github_token = event.flow.environment.integrations["github"].settings["api_token_secret"].retrieve_value()
+    headers = {"Authorization": f"Token {github_token}"}
+
+    all_options = []
+    per_page = 10
```
Comment (Arianna2028) [inline github_dynamic_targets/impl.py:24]: should we make this bigger so that we have to make less requests?

**Review by Arianna2028 — APPROVED** [review]

File: github_dynamic_targets/impl.py (hunk 3 above)
Comment (llam15) [reply github_dynamic_targets/impl.py:24]: Actually there's only 3 repos in the Sym Test organization l o l 😂

Comment (Arianna2028) [reply github_dynamic_targets/impl.py:24]: oh lol ok

**Pull request merged**